### ISSUE TITLE
Can't widen scope from typing.Callable to a callable class instance

### ISSUE THREAD

**Issue by FasterSpeeding** [microsoft/pyright#2079]
**Describe the bug**
Using an instance check to check if a variable type hinted as `Callable` is an instance of a class with a matching signature for it's `__call__` method is treated as a case which will will always fail even if it is valid.

**To Reproduce**

* Run pyright over the following code with strict type checking optionally enabled
* Watch it reveal "bar" to be "Never" within the if statement's scope

**Expected behavior**
bar's new type should be the type it was instance checked against within the if statement's scope.

**Screenshots or Code**
```py
import typing


class Foo:
    def __call__(self, arg: int, bar: str) -> None:
        raise NotImplementedError


def check_foo(maybe_foo: typing.Callable[[int, str], None]) -> bool:
    if isinstance(maybe_foo, Foo):
        reveal_type(maybe_foo)  # reveals info: Type of "bar" is "Never"
        return True

    return False


foo: typing.Callable[[int, str], None] = Foo()  # no error/warning on strict
```

**VS Code extension or command-line**
Pyright command line tool version 1.1.156
Windows 10
pythonVersion 3.8
typeCheckingMode = "strict"

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This fix is included in pyright 1.1.57, which I just published. It will also be included in the next release of pylance.
